### PR TITLE
Add test for ToConfigMapData with a nil profile

### PR DESCRIPTION
**Pull request by magic-peach**
## Description
ToConfigMapData's nil profile guard (return an error rather than panic or marshal garbage) had no test exercising it, unlike every other branch in this function. This adds one.

Fixes # (issue)
N/A, no existing issue, found while looking for untested error paths.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

This is a test only change, none of the above categories quite fit; no production code is touched.

## How Has This Been Tested?

- [ ] Manual test
- [x] Auto test added

## Checklist

- [x] My code follows the style guidelines of this project
- [x] My commit messages are signed-off
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pkg/component/worker/config/config_test.go [hunk 1]
```diff
@@ -28,6 +28,12 @@ func TestToConfigMapData(t *testing.T) {
 	}
 }
 
+func TestToConfigMapData_NilProfile(t *testing.T) {
```
Comment (twz123) [inline pkg/component/worker/config/config_test.go]: Let's have this as a subtest in `TestToConfigMapData`.

Comment (magic-peach) [reply pkg/component/worker/config/config_test.go]: Done, moved it in as a subtest.

**Comment by magic-peach** [comment]
@twz123 I made the changes

**Comment by twz123** [comment]
Could you please squash your changes?

**Comment by magic-peach** [comment]
@twz123 Squashed, thanks.

**Review by twz123 — APPROVED** [review]
